### ISSUE TITLE
Update to rfc8607 following the webdav specification.

### ISSUE THREAD

**Issue by masx200** [perry-mitchell/webdav-client#209]
Update to rfc8607 following the webdav specification.

https://www.ietf.org/rfc/rfc8607.html

So what functions does the current webdav-client support, and which functions does it not support?

I searched webdav on ietf's datatracker and found rfc as follows.

from rfc 2291 to rfc 8607.

https://datatracker.ietf.org/doc/search?name=webdav&sort=&rfcs=on&activedrafts=on&by=group&group=

I found that webdav has updated many new extension methods and headers and others.

However, some webdav servers or client have incomplete support for these functions.

The release date of these rfc is from February 1998 to June 2019.

I want to know whether there is currently a server or client that implements all webdav functions?

**Comment by perry-mitchell** [comment]
> Update to rfc8607 following the webdav specification.

That RFC appears to be for CalDAV, right? Not sure it's relevant here.

> I want to know whether there is currently a server or client that implements all webdav functions?

Most likely not - or at least not this one. Despite the name this library aims to provide, primarily, access to files via the WebDAV protocol. I like the idea of adding more common and useful WebDAV features - Being feature complete is a nice **goal**. That being said, it's far from that now, and the addition of new features is highly dependent on people requesting/adding them. There'll also be a number of functions here that wrap several webdav features into one. Think of this library as more of a toolkit with which to use to interact with WebDAV services - not an RFC-complete client.

tldr; This library doesn't implement all of the popular RFC features, but it could do. Naming schemes are my own. Will accept help on adding new features. Will add new features myself if enough requests come in for a particular feature, or if the feature is useful to my work as well.
